doc/3.1-create-selectors-objects.md

```diff
@@ -0,0 +1,103 @@
+
+To understand how works selectors objets and the function `selector()`, try some of the functions listed below and then debug your selector variable to see how it looks like
+
+### The list of the extensions possible
+Let suppose we have a `$btn--selector` variable where we had already initiated a selector object with the function `selector(new 'btn')`.
+
+In the table below we list all the possibilty to extend this selector object. Keep in mind that the `selector()` must be stocked into a variable.
+
+| Function | role |
+| -------- | ---- |
+| `selector(set $btn--selector prefix 'd')` | set the selector prefix |
+| `selector(set $btn--selector prefix-separator '-')` | set the selector prefix separator |
+| `selector(set $btn--selector block 'btn')` | set the selector block following BEM philosophy |
+| `selector(set $btn--selector tag 'a')` | add to the selector an html tag to specify |
+| `selector(set $btn--selector pseudo-classes ('hover', 'focus'))` | add to the selector a list of pseudo-classes to specify |
+| `selector(append $btn--selector pseudo-class 'hover')` | add to the selector a single pseudo-class to specify |
+| `selector(set $btn--selector element 'icon')` | add to the selector class an element (according to the BEM philosophy) |
+| `selector(set $btn--selector element-separator '__')` | set the selector element separator |
+| `selector(set $btn--selector modifiers ('primary', 'secondary'))` | add to the selector class a list of modifiers (according to the BEM philosophy) |
+| `selector(append $btn--selector modifier 'primary')` | add to the selector class a single modifier (according to the BEM philosophy) |
+| `selector(set $btn--selector modifier-separator '--')` | set the selector modifier separator |
+| `selector(set $btn--selector modifiers-element ('lg', 'sm'))` | add to the selector class a list of modifiers-element (according to the BEM philosophy) |
+| `selector(append $btn--selector modifier-element 'primary')` | add to the selector class a single modifier-element (according to the BEM philosophy) |
+| `selector(set $btn--selector attributes ('disabled', 'title^="disabled"'))` | add to the selector a list of attributes to specifiy |
+| `selector(append $btn--selector attribute 'disabled')` | add to the selector a single attribute to specify |
+| `selector(set $button--selector breakpoint 'sm')` | add to the selector a responsive breakpoint in order to generate a media-query bracing the selector based on the min-width of the breakpoint |
+| `selector(set $button--selector pseudo-element 'before')` | add to the selector a pseudo element to specify |
+| `selector(append $button--selector parent $button-group--selector as '>')` | add to the selector an another parent selector written just before, and its relationship |
+
```

Comment:
some selectors properties like "modifiers", "attributes" or "parents" are a collection of several values. That is why there "set" and "append" :

* "set" redefine the whole property collection
* "append" just add a new item into the collection

That's why we have to keep the both, but maybe we have to explain this difference between those two keywords.